### PR TITLE
Added events for expansion state changes for DtTreeTable

### LINKED ISSUE #415: Provide a toggle event handler for dt-tree-table-toggle-cell
# Feature Request

## Summary

Add a toggle event for collapsing or expanding the toggle cell in the dt-tree-table component.

## Feature Description
As there is no other way to detect that the toggle cell in the tree table has expanded or collapsed, a click handler on the entire toggle cell was used. But this solution isn't great as a click handler on the toggle icon (toggle event) would be way more precise and avoid weird behavior for the user.

### PR DESCRIPTION
**Pull request by gselltho**
Previously there was no way to figure out, if a tree has expanded or collapsed.
So `expandChanged(bool)`, `expanded()`, `collapsed()` events where introduced.
Expanded can also be bound with two way binding to set the initial state.

Fixes #415

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: components/tree-table/src/tree-table.spec.ts [hunk 1]
```diff
@@ -139,6 +139,77 @@ describe('DtTreeTable', () => {
       ]);
     });
 
+    // tslint:disable-next-line: dt-no-focused-tests
```
Comment (tomheller) [inline components/tree-table/src/tree-table.spec.ts]:
```suggestion
```

**Comment by lukasholzer** [comment]
Nice work some little coding standards like annotating internal functions with the correct jsdoc comment and using brackets :)

File: components/tree-table/src/tree-table-toggle-cell.ts [hunk 2]
```diff
@@ -61,6 +63,45 @@ export class DtTreeTableToggleCell<T> extends DtCell
...
+        this._expand();
+      } else {
+        this._collapse();
+      }
+    }
+  }
+
+  /** @internal Event emitted when the panel is expanded. */
+  @Output('expanded')
+  readonly _treeExpanded: Observable<boolean> = this.expandChange.pipe(
```
Comment (lukasholzer) [inline components/tree-table/src/tree-table-toggle-cell.ts]: Please remove @internal and you only have one expanded so there is no need for renaming the variable.

```suggestion
  @Output()
  readonly expanded: Observable<boolean> = this.expandChange.pipe(
```

File: components/tree-table/src/tree-table.spec.ts [hunk 3]
```diff
@@ -400,6 +484,19 @@ class SimpleDtTreeTableApp {
     return node;
   };
 
+  expandStateChanged(): void {
+    this.toggleCounter++;
+  }
+
+  hasCollapsed(row: any): void {
+    console.log('collapsed ' + JSON.stringify(row));
```
[the remark this answers is not in the document]
Comment (lukasholzer) [reply components/tree-table/src/tree-table.spec.ts]: And please remove this line :)

**Comment by ffriedl89** [comment]
I will take care of finalising the PR

**Comment by ffriedl89** [comment]
Finalised with #500

**Pull request closed without merging**